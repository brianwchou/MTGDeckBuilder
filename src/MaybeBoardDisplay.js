import React from 'react';
import Card from './components/Card/Card';
import { connect } from 'react-redux';

const mapStateToProps = (state) => {
    return {
        cards: state.maybeBoard.maybeBoardCards
    }
}

const maybeBoardStyle = {
    display: 'flex',
    flexWrap: 'wrap',
    border: 'solid black thin',
    width: '63vw',
    height: '28vh',
    overflow: 'scroll',
}

class MaybeBoardDisplay extends React.Component {
    constructor() {
        super();
    }

    render() {

        const cards = this.props.cards.map((info, index) => {
            return <Card info={info} type={"Clickable"} key={index} />
        })

        return(
            <div style={maybeBoardStyle}>
                {cards}
            </div>
        )
    }




}


export default connect(mapStateToProps)(MaybeBoardDisplay)