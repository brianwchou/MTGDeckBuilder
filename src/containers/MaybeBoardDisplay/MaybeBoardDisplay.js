import React from 'react';
import Card from '../../components/Card';
import { connect } from 'react-redux';

const mapStateToProps = (state) => {
    return {
        cards: state.maybeBoard.maybeBoardCards
    }
}

const maybeBoardStyle = {
    display: 'flex',
    flexWrap: 'wrap',
    border: 'solid black thin',
    width: '63vw',
    height: '28vh',
    overflow: 'scroll',
}

class MaybeBoardDisplay extends React.Component {
    render() {
        const cards = this.props.cards.map((info, index) => {
            return <Card info={info} key={index} />
        })

        return(
            <div style={maybeBoardStyle}>
                {cards}
            </div>
        )
    }
}


export default connect(mapStateToProps)(MaybeBoardDisplay)