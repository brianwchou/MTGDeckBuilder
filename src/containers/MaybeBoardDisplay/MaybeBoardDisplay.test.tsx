import { MaybeBoardDisplay, mapStateToProps } from 'containers/MaybeBoardDisplay/MaybeBoardDisplay'
import React from 'react'
import renderer from 'react-test-renderer'
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('MaybeBoardDisplay', () => {

    const dummyCard = {name:"dummy", image_uris:{small:"https://via.placeholder.com/300"}}
    const mockCards = [dummyCard]

    test('mapStateToProps', () => {
        const mockCards = [{name: "opt"}]
        const mockState = {maybeBoard: {cards: mockCards}}

        expect(mapStateToProps(mockState)).toEqual({ cards: [ { name: 'opt' } ] })
    })

    test('rendering cards correctly', () => {
        const tree = renderer
            .create(<MaybeBoardDisplay cards={mockCards}  />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })

    test('getCardInfo dispatches addToDecklist', () => {      
        const onClickSpy = jest.fn()
        const wrapper = mount(<MaybeBoardDisplay cards={mockCards} dispatch={onClickSpy} />)

        const mockTarget = {target: { name: "add"} }
        let container = wrapper.find("div")
        let containerButton = container.find("button").at(0)

        expect(onClickSpy).not.toHaveBeenCalled();
        containerButton.simulate('click', mockTarget);
        expect(onClickSpy).toHaveBeenCalled();
    })

    test('getCardInfo dispatches deleteFromMaybe', () => { 
        const onClickSpy = jest.fn()
        const wrapper = mount(<MaybeBoardDisplay cards={mockCards} dispatch={onClickSpy}  />)
        
        const mockTarget = {target: { name: "other"} }
        let container = wrapper.find("div")
        let containerButton = container.find("button").at(1)

        expect(onClickSpy).not.toHaveBeenCalled();
        containerButton.simulate('click', mockTarget);
        expect(onClickSpy).toHaveBeenCalled();        
    })

    test('getCardInfo does nothing when button name is neither add or other', () => {
        const onClickSpy = jest.fn()
        const wrapper = mount(<MaybeBoardDisplay cards={mockCards} dispatch={onClickSpy}  />)
        
        const mockTarget = {target: { name: ""} }
        let container = wrapper.find("div")
        let containerButton = container.find("button").at(0)

        expect(onClickSpy).not.toHaveBeenCalled();
        containerButton.simulate('click', mockTarget);
        expect(onClickSpy).not.toHaveBeenCalled();     
    })
})