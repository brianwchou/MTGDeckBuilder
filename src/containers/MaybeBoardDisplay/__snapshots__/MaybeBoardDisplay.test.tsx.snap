// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaybeBoardDisplay rendering cards correctly 1`] = `
<div
  style={
    Object {
      "border": "solid black thin",
      "display": "flex",
      "flexWrap": "wrap",
      "height": "30vh",
      "overflow": "scroll",
      "width": "50vw",
    }
  }
>
  <div
    className="container"
  >
    <img
      alt="dummy"
      draggable={false}
      src="https://via.placeholder.com/300"
    />
    <button
      className="btn"
      name="add"
      onClick={[Function]}
    >
      Add To Decklist
    </button>
    <button
      className="btn2"
      name="other"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
